frontend/src/contexts/AuthContext.tsx

```diff
@@ -0,0 +1,70 @@
+import Keycloak from 'keycloak-js';
+import {
+  createContext,
+  FC,
+  PropsWithChildren,
+  useEffect,
+  useState,
+} from 'react';
+import {
+  REACT_APP_CROWNLABS_OIDC_PROVIDER_URL,
+  REACT_APP_CROWNLABS_OIDC_CLIENT_ID,
+} from '../env';
+interface IAuthContext {
+  isLoggedIn: undefined | boolean;
+  token: undefined | string;
+  userId: string | undefined;
```

Comment:
Isn't something like this working? 
```suggestion
  isLoggedIn?: boolean;
  token?: string;
  userId?: string;
```
If not, couldn't we still use empty strings to say those values aren't present?
(also do we need isLoggedIn to be undefined?)

Reply:
We use `undefined` for 2 reasons:
- with the isLoggedin, at first you don't actually know if the user is logged in or not, so I used it there, to hva more control. Although, in the case of our app, the user can't do anything if not logged in, so it could be a simple boolean, where the false means "unknown or not logged in"
- for the other cases I prefer to have undefined instead of an empty string to have a more explicit code. And yes, I think we can use that syntax